rewrite-maven/src/main/java/org/openrewrite/maven/internal/MavenPomDownloader.java

```diff
@@ -665,6 +665,11 @@ public MavenRepository normalizeRepository(MavenRepository originalRepository, @
             if (repository.getUri().contains("${")) {
                 return null;
             }
+            // Skip blocked repositories
+            // See https://github.com/openrewrite/rewrite/issues/3141
+            else if (repository.getUri().contains("0.0.0.0")) {
+                return null;
+            }
```

Comment:
This solves the issue of blocked repositories, but does not fix that our retry configuration does not have a limit

```java

private static final RetryConfig retryConfig = RetryConfig.custom()
            .retryOnException(throwable -> throwable instanceof SocketTimeoutException ||
                                           throwable instanceof TimeoutException)
            .build();


```

We can configure a limit using `maxAttempts`, but I am just not sure what value I should set and that's why I did not include it with this change



```java

private static final RetryConfig retryConfig = RetryConfig.custom()
            .retryOnException(throwable -> throwable instanceof SocketTimeoutException ||
                                           throwable instanceof TimeoutException)
            .maxAttempts(<int>)
            .build();


```